### PR TITLE
Creating basic functionality to connect to Discord API, work in progress.

### PR DESCRIPTION
**Pull request by NotReallyEight**
- [x] Connect function.
- [x] Identify.
- [x] Interval to send heartbeat.
- [ ] Handling all kind of [events](https://discord.com/developers/docs/topics/gateway#commands-and-events-gateway-events) from Discord through OP Code.
- [x] Make the actual resume a function, so it can be used throughout the code.
- [x] Export Private Channel, Guild Channel from `structures/index.ts`.
- [ ] Listen to 'close' event in WebSocket, and handle close code 1006, as Connection closed abnormally
- [x] Update method for all the classes.
- [x] Make a `AnyGuildChannel` type alias with all the guild channel type definitions
- [x] Store `seq` as the `s` property in payloads, if it does not exist or it is null, then make it the last one it received.

### CODE REVIEW

**Comment by NotReallyEight** [comment]
Need to uninstall un pkg I installed by fault.

File: src/Client.ts [hunk 1]
```diff
@@ -32,6 +41,22 @@ export class Client extends EventEmitter {
 	 */
 	rest = new Rest(this);
 
+	/**
+	 * The last sequence number received from the WebSocket server
+	 */
+	seq: number = 0;
```
Comment (DTrombett) [inline src/Client.ts:47]:
```suggestion
	seq = 0;
```

File: src/structures/PrivateChannel.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import type Client from "../Client";
+import type {
+	APIChannel,
+	Snowflake,
+} from "../../node_modules/discord-api-types/v9";
```
Comment (DTrombett) [inline src/structures/PrivateChannel.ts:5]: Blame autoimport
```suggestion
} from "discord-api-types/v9";
```

File: src/structures/PrivateChannel.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	 * The ID of the last message sent in the channel
+	 */
+	lastMessageId: Snowflake | null | undefined;
+
+	/**
+	 * The recipients of the channel
+	 */
+	recipients?: User[];
+
+	constructor(payload: APIChannel, client: Client) {
```
Comment (DTrombett) [inline src/structures/PrivateChannel.ts:25]: 🧹
```suggestion
	constructor(client: Client, payload: APIChannel) {
```

File: src/structures/PrivateChannel.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import type Client from "../Client";
+import type {
+	APIChannel,
+	Snowflake,
+} from "../../node_modules/discord-api-types/v9";
+import { Channel } from "./Channel";
```
Comment (DTrombett) [inline src/structures/PrivateChannel.ts:6]:
```suggestion
import Channel from "./Channel";
```

File: src/structures/PrivateChannel.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	 * The recipients of the channel
+	 */
+	recipients?: User[];
+
+	constructor(payload: APIChannel, client: Client) {
+		super(payload, client);
+		this.client = client;
+		this.lastMessageId = payload.last_message_id;
+		this.recipients = payload.recipients?.map(
+			(recipient) => new User(recipient, client)
```
Comment (DTrombett) [inline src/structures/PrivateChannel.ts:30]:
```suggestion
			(recipient) => new User(client, recipient)
```

File: src/structures/User.ts [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+	/**
+	 * The public flags on a user's account
+	 */
+	publicFlags?: UserFlags;
+
+	/**
+	 * @param payload - The payload for the user
+	 * @param client - The client that instantiated this class
+	 */
+	constructor(payload: APIUser, client: Client) {
```
Comment (DTrombett) [inline src/structures/User.ts:93]:
```suggestion
	constructor(client: Client, payload: APIUser) {
```

**Review by DTrombett — CHANGES_REQUESTED** [review]

**Comment by DTrombett** [comment]
TODO:

- [ ] Add new Application properties (discordjs/discord-api-types#239)

**Comment by NotReallyEight** [comment]
I think, we should change this Pull Request, and move it to other separate pull requests. This PR is getting just way too big and it already does what it needs to do. I will merge it soon.

**Pull request merged**